MERLin/MERLin/Routing/RouterProtocol.swift

```diff
@@ -11,8 +11,32 @@ import RxSwift
 
 public typealias ViewControllersFactory = ViewControllerBuilding & DeeplinkManaging
 
-public protocol Routing {
-    var router: Router { get }
+public enum ContextUpdatableSearchPreference {
```

Comment:
It's better add description doc here for each case

Reply:
I'm not sure it's right. MERLin gives an implementation of these search preferences, but it is not THE implementation. This enum carries no promises around the algorithm to be used to accomplish these preferences. It really depends on the implementation where these are used. I can add more comments on the handle method, but it's always going to be something that just who digs into MERLin will be able to read as the handle method is internal.